### DISCUSSION TITLE
Cache file definable?

### DISCUSSION THREAD

**Discussion by gitttt54** [meisnate12/Plex-Meta-Manager#1266] [Q&A]
Hello,

I have 2 configs  (with more or less same content and libraries) and therefore 2 cache files.
Is it possible to tell PMM to only use 1 single cache file?
Or does it make no sense?

Thank you

**Comment by chazlarson** [comment]
There is currently no provision for specifying a cache file; it's tied to the config.  What's the problem you are trying to solve?  Chances are this wouldn't solve it in the way you are anticipating.

**Reply by gitttt54** [reply]
Thank you, there is no problem, 
I just thought "why not use one single cache file", because the stock of data is the same for both configs and therefore it could spare time and ressources not generating the cache file twice.

Not a big deal, because those files don't take much space, only creating time bothers me a little bit (about 20 mins I guess), but only when I'm in front of the computer, waiting for TMM to finish to see the results ;-D

**Reply by chazlarson** [reply]
Do those two config files point to two different servers?  If so, there wouldn't be complete overlap anyway since the Plex internal keys for the items would quite likely be different.

If they are pointing to the same plex server, the obvious solution is to combine the config files to unify the cache.

**Reply by YozoraXCII** [reply]
Without full context, if you want to run the same config file against more than one server, then you can use Run Commands to do this, reducing the need to have a different config file per server:
[https://metamanager.wiki/en/latest/home/environmental.html#env-plex-url-token](url)

You need to set your plex `url` and `token` within the config.yml to `ENV`

**Reply by gitttt54** [reply]
>Do those two config files point to two different servers?

no.

>If they are pointing to the same plex server, the obvious solution is to combine the config files to unify the cache.

Well, thats somehow OT, but since PMM runs scheduled (and that can not disabled), I have splitted my configs.

One, which has short stuff to test and my watchlist and some trending/popular lists which I run mostly by hand. (takes 30-50mins to finish).
The other one with stuff that takes very long, like genre mapping and other mass operations and dynamic collections, which runs scheduled (takes 12 hours to finish).

So combining is kinda cumbersome to me, because then I have to use "test: true" more often and also I have not figured out to skip time consuming things like mass operations or genre mappings within a test run and I don't want to wait hours to see results when testing.
Now, I just run `docker exec -it pmm python plex_meta_manager.py --run -c config/config_mm.yml --no-missing` and can see the results in 30 mins. Thats fine. 
The other configs runs when I sleep ;-D


Anyways, my original questions is answered. Thank you.

**Reply by meisnate12** [reply]
https://metamanager.wiki/en/latest/home/environmental.html

so you can use `--run-metadata-files` to pick which metadata file you wish to run 

`--collections-only` will run only collections no operations

**Reply by gitttt54** [reply]
> https://metamanager.wiki/en/latest/home/environmental.html
> 
> so you can use `--run-metadata-files` to pick which metadata file you wish to run
> 
> `--collections-only` will run only collections no operations

Yeah, its all possible, I assume,
but my brain can not remember all those options for the vary of situations I run in.
(Its like having a swiss pocketknife with all those fancy helpers, but ending up with using just the biggest knife and sometimes the toothpick to clean your fingernails)

I know it would be easy to just change some parameter in the command line and also add some "test: true" lines in the metadata files, but, really, thats not viable for me.

I'm from the ages, where we commented out things we don't wanted to be executed.

I know, it would be a straightforwarded way for me, and I tried it in the first months using PMM, but having 10 variations of command lines to test things (and a textfile to save them) is just not the way it works for me. (Also, more important, it presumes a full understanding of all those PMM parameters and the context they work in.)

So I split my configs and comment out.

Thank you so much for giving us all the possibilities! and I'm very sorry, I can not use them. PMM is very complex and my brain isn't (anymore).
